simul/cothority/cothority.go

```diff
@@ -77,25 +78,33 @@ func main() {
 		// If this cothority has the root-host, it will start the simulation
 		dbg.Lvl2("Starting protocol", simul, "on host", rootSC.Host.Entity.Addresses)
 		//dbg.Lvl5("Tree is", rootSC.Tree.Dump())
+
+		// First count the number of available children
 		childrenWait := monitor.NewMeasure("ChildrenWait")
 		wait := true
+		networkDelay := 500
+		timeout := networkDelay * 2 * int(math.Log2(float64(len(rootSC.EntityList.List))))
 		for wait {
-			dbg.Lvl2("Counting children")
-			node, err := rootSC.Overlay.StartNewNodeName("Count", rootSC.Tree)
+			dbg.Lvl2("Counting children with timeout of", timeout)
+			node, err := rootSC.Overlay.CreateNewNodeName("Count", rootSC.Tree)
 			if err != nil {
 				dbg.Fatal(err)
 			}
+			node.ProtocolInstance().(*manage.ProtocolCount).Timeout = timeout
+			node.ProtocolInstance().(*manage.ProtocolCount).NetworkDelay = networkDelay
+			node.Start()
 			select {
 			case count := <-node.ProtocolInstance().(*manage.ProtocolCount).Count:
 				if count == rootSC.Tree.Size() {
-					dbg.Lvl2("Found all", count, "children")
+					dbg.Lvl1("Found all", count, "children")
 					wait = false
 				} else {
 					dbg.Lvl1("Found only", count, "children, counting again")
 				}
-			case <-time.After(time.Second * 10):
+			case <-time.After(time.Millisecond * time.Duration(timeout) * 2):
```

Comment:
Can we have short comments why the time(out) is like that (why `* 2` because most of the time it is a binary tree, or does it have other reasons?)